supplementary/alica_ros1/alica_tests/src/test/test_success.cpp

```diff
@@ -0,0 +1,110 @@
+#include <alica/test/TestContext.h>
+#include <alica/test/Util.h>
+#include <alica_tests/BehaviourCreator.h>
+#include <alica_tests/ConditionCreator.h>
+#include <alica_tests/ConstraintCreator.h>
+#include <alica_tests/PlanCreator.h>
+#include <alica_tests/TestWorldModel.h>
+#include <alica_tests/TransitionConditionCreator.h>
+#include <alica_tests/UtilityFunctionCreator.h>
+#include <clock/AlicaRosTimer.h>
+#include <communication/AlicaDummyCommunication.h>
+#include <logger/AlicaRosLogger.h>
+
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+namespace alica::test
+{
+
+class TestSuccessFixture : public ::testing::Test
+{
+public:
+    virtual void SetUp() override
+    {
+        ros::NodeHandle nh;
+        std::string path;
+        nh.param<std::string>("/rootPath", path, ".");
+        _tc = std::make_unique<TestContext>("hairy", path + "/etc/", "Roleset", "TestMasterPlan", true, 1);
+        ASSERT_TRUE(_tc->isValid());
+        const YAML::Node& config = _tc->getConfig();
+        _spinner = std::make_unique<ros::AsyncSpinner>(config["Alica"]["ThreadPoolSize"].as<int>(4));
+        _tc->setCommunicator<alicaDummyProxy::AlicaDummyCommunication>();
+        _tc->setTimerFactory<alicaRosTimer::AlicaRosTimerFactory>();
+        _tc->setLogger<alicaRosLogger::AlicaRosLogger>(config["Local"]["ID"].as<int>());
+        _tc->setWorldModel<alicaTests::TestWorldModelNew>(_tc.get());
+        AlicaCreators creators{std::make_unique<alica::ConditionCreator>(), std::make_unique<alica::UtilityFunctionCreator>(),
+                std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>(), std::make_unique<alica::PlanCreator>(),
+                std::make_unique<alica::TransitionConditionCreator>()};
+        _tc->init(std::move(creators));
+        _tc->startEngine();
+        _spinner->start();
+        STEP_UNTIL2(_tc, _tc->getActivePlan("TestMasterPlan"));
+        ASSERT_TRUE(_tc->getActivePlan("TestMasterPlan"));
+    }
+
+    virtual void TearDown() override
+    {
+        _spinner->stop();
+        _tc->terminate();
+    }
+
+protected:
+    std::unique_ptr<TestContext> _tc;
+    std::unique_ptr<ros::AsyncSpinner> _spinner;
+};
+
+TEST_F(TestSuccessFixture, behSuccessTest)
+{
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BehSuccessTestState"));
+    STEP_UNTIL2(_tc, _tc->getActiveBehaviour("SuccessOnInitBeh"));
```

Comment:
You should be able to use STEP_UNTIL here, when you pass the context to the makro, it will automatically forward it to STEP_UNTIL2